tee-worker/litentry/core/assertion-build/src/dynamic/contracts/token_holding_amount/TokenQueryLogic.sol

```diff
@@ -47,46 +48,111 @@ abstract contract TokenQueryLogic is TokenHoldingAmount {
 		if (identityToStringSuccess) {
 			uint256 totalBalance = 0;
 
-			string memory tokenContractAddress = tokenAddresses[tokenName][
+			string memory tokenContractAddress = getTokenAddress(
+				tokenName,
 				network
-			];
+			);
+
 			if (GeniidataClient.isSupportedNetwork(network)) {
 				uint256 balance = GeniidataClient.getTokenBalance(
-					secrets,
+					secrets[0],
 					identityString,
 					tokenName,
 					getTokenDecimals()
 				);
 				totalBalance += balance;
 			} else if (NoderealClient.isSupportedNetwork(network)) {
 				(bool success, uint256 balance) = NoderealClient
-						.getTokenBalance(
+					.getTokenBalance(
 						network,
-						secrets,
+						secrets[1],
 						tokenContractAddress,
 						identityString
 					);
 				if (success) {
 					totalBalance += balance;
 				}
+			} else if (MoralisClient.isSupportedNetwork(network)) {
+				if (Strings.equal(tokenContractAddress, "Native Token")) {
+					(
+						bool success,
+						string memory solanaTokenBalance
+					) = MoralisClient.getSolanaNativeBalance(
+							network,
+							secrets[2],
+							identityString
+						);
+
+					if (success) {
+						(bool parsedStatus, uint256 parsedAmount) = Utils
+							.parseDecimal(
+								solanaTokenBalance,
+								getTokenDecimals()
+							);
+						if (parsedStatus) {
+							totalBalance += parsedAmount;
+						}
+					}
+				} else {
+					(
+						bool success,
+						SolanaTokenBalance[] memory solanaTokenBalance
+					) = MoralisClient.getSolanaTokensBalance(
+							network,
+							secrets[2],
+							identityString
+						);
+
+					if (success) {
+						for (uint i = 0; i < solanaTokenBalance.length; i++) {
+							if (
+								Strings.equal(
+									solanaTokenBalance[i].mint,
+									tokenAddresses[tokenName][network]
```

Comment:
should we use tokenContractAddress here?

Reply:
thanks, my bad,careless.